snooty/language_server.py

```diff
@@ -259,6 +259,14 @@ def m_initialized(self, **kwargs: object) -> None:
         # Ignore this message to avoid logging a pointless warning
         pass
 
+    def m_text_document__resolve(self, path: str) -> str:
+        """Pass a path from a document using method 'textDocument/resolve' and return the full path"""
+
+        if self.project is None:
+            return self.root_uri + path
+
+        return f"{self.project.config.source_path}" + path
```

Comment:
This use of an f-string seems unnecessary.

source_path is a Path, so you'd just want str(server.project.config.source_path.joinpath(path)), I think?